pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -2573,66 +2573,37 @@ public Set<String> getSegmentsYetToBeCommitted(String tableNameWithType, Set<Str
    * @param committingSegments List of new segment names that are currently in COMMITTING state.
    *                          If null, returns true without making any changes to the existing list
    * @return true if the synchronization succeeds, false if there's a failure in updating ZooKeeper
-   * @see #getCommittingSegments for the logic that filters out segments no longer in COMMITTING state
    */
-  public boolean syncCommittingSegments(String realtimeTableName, @NotNull List<String> committingSegments) {
+  public boolean syncCommittingSegments(String realtimeTableName, List<String> committingSegments) {
```

Comment:
Is this method thread safe? What happens when we get two parallel calls to this method with same table but different comittingSegments?

Reply:
It should be thread safe because we perform version check before posting the metadata to ZK